system-addon/lib/TelemetrySender.jsm

```diff
@@ -0,0 +1,126 @@
+/* globals Preferences, Services, XPCOMUtils */
+
+const {interfaces: Ci, utils: Cu} = Components;
+
+Cu.import("resource://gre/modules/Preferences.jsm");
+Cu.importGlobalProperties(["fetch"]);
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+const ENDPOINT_PREF = "telemetry.ping.endpoint";
+const TELEMETRY_PREF = "telemetry";
+const LOGGING_PREF = "performance.log";
+
+const ACTION_NOTIF = "user-action-event";
+const PERFORMANCE_NOTIF = "performance-event";
+const COMPLETE_NOTIF = "tab-session-complete";
```

Comment:
I'd propose we land as-is and adjust when we do our performance event audit in or after that meeting.  Or Kate can adjust as necessary in her TabSender patch.